### PR TITLE
EXP-276: Implementar repasse de taxas na criação de link de pagamento

### PR DESCRIPTION
**Pull request by diegodsgarcia**
## Contexto
Adiciona criação de links com o repasse de taxas para clientes link.me

## Checklist
- [x] Adiciona utils `hasAnticipation `e `canChargeTransactionFee`
- [x] Adiciona repasse de taxas
- [x] Atualiza página de detalhes do link com as informações de parcelamento de acordo com repasse de taxas
- [x] Adiciona addon knobs no storybook
- [x] Atualiza story `PaymentLinkAdd` com o addon canChargeTransectionFee

## Issues linkadas
- [x] [EXP-276: implementar repasse de taxas na criação de link de pagamento](https://mundipagg.atlassian.net/secure/RapidBoard.jspa?rapidView=185&projectKey=EXP&modal=detail&selectedIssue=EXP-276)

### Layout:
https://www.figma.com/file/0QOwU9BIcrLOwCHz2xurA4/Link.me?node-id=4167%3A729

### Preview:

**DESCRIÇÃO DO LINK**
Screen Shot 2020-09-09 at 13 50 26
Screen Shot 2020-09-09 at 13 51 07
Screen Shot 2020-09-09 at 13 53 12

## Como testar?
- Faça login em uma company link.me com o repasse de taxas ativa
- Realize a criação de link com fluxo de repasse de taxas tanto do cliente quanto do lojista
- Entre na página de descrição de link e veja as informações de parcelamento do cartão
- Crie um link tbm em um cliente sem repasse de taxas.
- Verifique os story do PaymentLinks > List > PaymentLinkAdd

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f271e114c4e53ef371f8847248983d591ca3cad7
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!  


 * Commit: cb2d1f4429cd448e0361b3ade700cafe2bb838d1
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f271e114c4e53ef371f8847248983d591ca3cad7
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!  


 * Commit: cb2d1f4429cd448e0361b3ade700cafe2bb838d1
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f271e114c4e53ef371f8847248983d591ca3cad7
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!  


 * Commit: cb2d1f4429cd448e0361b3ade700cafe2bb838d1
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f271e114c4e53ef371f8847248983d591ca3cad7
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!  


 * Commit: cb2d1f4429cd448e0361b3ade700cafe2bb838d1
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 131c830d3554250deb9be8f70a0ff4315681c85b
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 131c830d3554250deb9be8f70a0ff4315681c85b
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 131c830d3554250deb9be8f70a0ff4315681c85b
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 131c830d3554250deb9be8f70a0ff4315681c85b
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by allangrds** [comment]
Não esquecer de atualizar o `package-lock.json`, por favor.

**Comment by allangrds** [comment]
Tá show, mano. Testei tudo. Só resolver o que falei acima e tá blz.

File: packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js [hunk 1]
```diff
@@ -25,6 +26,49 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
...
+                from: freeInstalments + 1,
+                to: maxInstallments,
+              })}
+            </span>
+            {t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_1')}
+          </>
+        )}
+      </>
+    )
+  }
```
Comment (vagnervst) [inline packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js:70]: acho que pode mudar esse bloco pra evitar a repetição de código entre esse if e o de cima:

```diff
diff --git a/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js b/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js
index 11e6450e..124a7c5d 100644
--- a/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js
+++ b/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js
@@ -26,34 +26,21 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
     )
   }
 
-  if (chargeTransactionFee && freeInstalments === 0) {
+  if (chargeTransactionFee) {
     return (
       <>
         {t('pages.payment_link_detail.payment_methods.with_interest_rate_2')}
         <span>
           {t('pages.payment_link_detail.payment_methods.to', {
             from: 1,
-            to: maxInstallments,
-          })}
-        </span>
-        {t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_1')}
-      </>
-    )
-  }
-
-  if (chargeTransactionFee && freeInstalments > 0) {
-    return (
-      <>
-        {t('pages.payment_link_detail.payment_methods.with_interest_rate_2')}
-        <span>
-          {t('pages.payment_link_detail.payment_methods.to', {
-            from: 1,
-            to: freeInstalments,
+            to: freeInstalments > 0
+              ? freeInstalments
+              : maxInstallments,
           })}
         </span>
         {t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_2')}
         <br />
-        {freeInstalments < 12 && (
+        {freeInstalments > 0 && freeInstalments < 12 && (
           <>
             {t('pages.payment_link_detail.payment_methods.with_interest_rate_2')}
             <span>

```

File: packages/pilot/package.json [hunk 2]
```diff
@@ -51,6 +51,7 @@
     "@percy-io/in-percy": "0.1.11",
     "@storybook/addon-a11y": "5.2.5",
     "@storybook/addon-actions": "5.2.5",
+    "@storybook/addon-knobs": "^6.0.21",
```
[the remark this answers is not in the document]
Comment (diegodsgarcia) [reply packages/pilot/package.json:54]: Estamos a 0 PR's sem eu esquecer desse detalhe huahuashuashusahusq. Vlwww Vaguinho

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2e58f4eec12b6722416f56c705ee7206ce8e10d7
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2e58f4eec12b6722416f56c705ee7206ce8e10d7
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2e58f4eec12b6722416f56c705ee7206ce8e10d7
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2e58f4eec12b6722416f56c705ee7206ce8e10d7
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by allangrds** [comment]
@diegodsgarcia não commitou o `package-lock`.

**Comment by diegodsgarcia** [comment]
> @diegodsgarcia não commitou o `package-lock`.

Mas a gente não usa o yarn na pilot? 🤔 .

File: packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js [hunk 3]
```diff
@@ -25,6 +26,39 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
     )
   }
 
+  if (chargeTransactionFee) {
```
Comment (vagnervst) [inline packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js:29]: repensando um pouco a forma como mostramos essas parcelas, da pra imaginar cada linha como um objeto e fazer um loop, ve o que acha:

```jsx
if (chargeTransactionFee) {
    const installmentsGroups = freeInstalments > 0
      ? [{from: 1, to: freeInstalments}, {from: freeInstalments, to: maxInstallments}]
      : [{from: 1, to: maxInstallments}]

    return (
      <>
        {installmentsGroups.map(({ from, to }) => (
          <>
            {t('pages.payment_link_detail.payment_methods.with_interest_rate_2')}
            <span>
              {t('pages.payment_link_detail.payment_methods.to', {
                from,
                to,
              })}
            </span>
            {to !== maxInstallments
              ? t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_2')
              : t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_1')
            }
            <br />
          </>
        ))}
      </>
    )
  }
```

File: packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js [hunk 4]
```diff
@@ -25,6 +26,38 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
...
+        { from: 1, to: maxInstallments },
+      ]
+
+    return (
+      <>
+        {installmentsGroup.map(({ from, to }) => (
+          <Fragment key={from}>
+            {t('pages.payment_link_detail.payment_methods.with_interest_rate_2')}
+            <span>
+              {t('pages.payment_link_detail.payment_methods.to', {
```
Comment (vagnervst) [inline packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js:45]: notei que temos um problema com pluralidade nessa translation quando free_installments === 1:

image

para resolver isso podemos usar o `count` do i18n para decidir entre a translation com ou sem pluralidade. Abaixo a diff que fiz com as mudanças que corrigem isso, que consiste de

1. adicionei variações com singularidade nas translations para tratar o caso de uma parcela sem cobrança
2. alterei os estilos para não aplicar margin em todo `span`, isso se fez necessário pois o "De" de "De x até y vezes" agora nem sempre aparece, mas a margem era sempre aplicada no `span` para criar o espaçamento entre "De" e o número
3. adicionei uma classe nova para manter a margin em volta de todo `span` no caso de link de pagamento com `interest_rate`, pois necesse cenário colocamos os números sempre em bold e em um `span`, fazendo necessário esse espaçamento

<details>
<summary>Diff com o fix de pluralidade</summary>


```diff
diff --git a/packages/pilot/public/locales/pt/translations.json b/packages/pilot/public/locales/pt/translations.json
index ac1fcc3a..3303883e 100644
--- a/packages/pilot/public/locales/pt/translations.json
+++ b/packages/pilot/public/locales/pt/translations.json
@@ -1828,10 +1828,13 @@
         "installment": "parcela",
         "installment_plural": "parcelas",
         "with_interest_rate_2": "De",
-        "to": "{{from}} a {{to}}",
+        "to": "Até {{to}}",
+        "to_plural": "{{from}} a {{to}}",
         "with_interest_rate_3": "vezes, juros de {{percent}} a.m",
-        "charge_transaction_fee_interest_rate_taxing": "vezes, com juros",
-        "charge_transaction_fee_interest_rate_no_taxing": "vezes, sem juros"
+        "charge_transaction_fee_interest_rate_taxing": "vez, com juros",
+        "charge_transaction_fee_interest_rate_taxing_plural": "vezes, com juros",
+        "charge_transaction_fee_interest_rate_no_taxing": "vez, sem juros",
+        "charge_transaction_fee_interest_rate_no_taxing_plural": "vezes, sem juros"
       },
       "list": {
         "empty_state_title": "Não foram encontradas transações neste link de pagamentos.",
diff --git a/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js b/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js
index 42a8bb31..774488a3 100644
--- a/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js
+++ b/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js
@@ -40,17 +40,28 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
       <>
         {installmentsGroup.map(({ from, to }) => (
           <Fragment key={from}>
-            {t('pages.payment_link_detail.payment_methods.with_interest_rate_2')}
-            <span>
+            {to > 1 && (
+              <span className={styles.from}>
+                {t('pages.payment_link_detail.payment_methods.with_interest_rate_2')}
+              </span>
+            )}
+            <span className={styles.range}>
               {t('pages.payment_link_detail.payment_methods.to', {
+                count: to,
                 from,
                 to,
               })}
             </span>
-            {to > freeInstalments && freeInstalments < 12
-              ? t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_taxing')
-              : t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_no_taxing')
-            }
+            <span className={styles.to}>
+              {to > freeInstalments && freeInstalments < 12
+                ? t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_taxing', {
+                  count: to,
+                })
+                : t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_no_taxing', {
+                  count: to,
+                })
+              }
+            </span>
             <br />
           </Fragment>
         ))}
@@ -59,7 +70,7 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
   }
 
   return (
-    <>
+    <div className={styles.interestRate}>
       {t('pages.payment_link_detail.payment_methods.max_of')}
       <span>{maxInstallments}</span>
       {t('pages.payment_link_detail.payment_methods.installment', { count: maxInstallments })}
@@ -72,6 +83,7 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
       {t('pages.payment_link_detail.payment_methods.with_interest_rate_2')}
       <span>
         {t('pages.payment_link_detail.payment_methods.to', {
+          count: maxInstallments,
           from: freeInstalments + 1,
           to: maxInstallments,
         })}
@@ -79,7 +91,7 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
       {t('pages.payment_link_detail.payment_methods.with_interest_rate_3', {
         percent: percent(interestRate),
       })}
-    </>
+    </div>
   )
 }
 
diff --git a/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/styles.css b/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/styles.css
index e276e038..bfefa394 100644
--- a/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/styles.css
+++ b/packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/styles.css
@@ -24,8 +24,22 @@
   color: var(--color-birdperson-gray-500);
   font-size: 16px;
 
-  & span {
+  & .from {
+    margin-right: 4px;
+  }
+
+  & .range {
     font-weight: bold;
-    margin: 0 4px;
+  }
+
+  & .to {
+    margin-left: 4px;
+  }
+
+  & > .interestRate {
+    & span {
+      margin: 0 4px;
+      font-weight: bold;
+    }
   }
 }
```
</details>

**Review by vagnervst — COMMENTED** [review]
só alguns detalhes e o problema com pluralidade, de resto ta show!

**Comment by ThaisRobba** [comment]
@diegodsgarcia Pilot é yarn mesmo, porque monorepo.
Reparei que a story de macrosegment do onboarding está quebrando mas parece que ela já estava quebrada, só que agora ficou quebrada diferente xD Não sei se entra no escopo dessa pr então mas acho que vale uma olhadinha, porque parece que ela usa um componente que foi modificado aqui.

**Comment by diegodsgarcia** [comment]
> @diegodsgarcia Pilot é yarn mesmo, porque monorepo.
> Reparei que a story de macrosegment do onboarding está quebrando mas parece que ela já estava quebrada, só que agora ficou quebrada diferente xD Não sei se entra no escopo dessa pr então mas acho que vale uma olhadinha, porque parece que ela usa um componente que foi modificado aqui.

Não tinha a ver, porém eu corrigi mesmo assim pq tava totalmente quebrado mesmo por coisa simples kkkkk

File: packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js [hunk 5]
```diff
@@ -25,8 +26,56 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
...
+              {to > freeInstalments && freeInstalments < 12
+                ? t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_taxing', {
+                  count: to,
+                })
+                : t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_no_taxing', {
+                  count: to,
+                })
+              }
+            </span>
+            <br />
```
Comment (ThaisRobba) [inline packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js:70]: Aqui talvez fizesse sentido usar um `<p>` ao invés dum fragmento com spans, até porque a gente acaba precisando dum espaço embaixo (isso provavelmente permitiria a gente remover o `</ br>`, condensar os estilos e tal).

Poderia ficar algo como

```javascript
<p key={from} className={styles.installmentRange}> 
  {to > 1 && t('pages.payment_link_detail.payment_methods.with_interest_rate_2')}
  <b>{t('pages.payment_link_detail.payment_methods.to', {
      count: to,
      from,
      to,
    })}
  </b>
  {to > freeInstalments && freeInstalments < 12
                ? t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_taxing', {
                  count: to,
                })
                : t('pages.payment_link_detail.payment_methods.charge_transaction_fee_interest_rate_no_taxing', {
                  count: to,
                })
              }
</p>
```

O que tu acha?

File: packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js [hunk 6]
```diff
@@ -25,8 +26,56 @@ const renderCreditCardMessage = (creditCardConfig, t) => {
     )
   }
 
+  if (chargeTransactionFee) {
```
Comment (ThaisRobba) [inline packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js:29]: Acho que talvez vale a pena mudar a separação um teco, porque ficaram dois valores de retorno de tamanho considerável. Dai o que está sendo retornado poderia ser convertido num componente (acho que interno mesmo do container, num precisa mover pra fora) e o if poderia ficar algo como:

```
return chargeTransactionFee
  ? <PaymentMethodWithTransactionFee />
  : <PaymentMethodWithoutTransactionFee />
```

O que tu acha? :)

**Comment by ThaisRobba** [comment]
@diegodsgarcia Dá uma olhadinha nos stories de `Containers|Payment Links/Details/PaymentMethods: WithInterestRate`
e `Containers|Payment Links/Details/PaymentMethods: WithoutInterestRate` que ficou meio peculiar as mudanças 🤔

File: packages/pilot/src/containers/PaymentLinks/List/PaymentLinkAdd/SecondStep/CreditCardInput.js [hunk 7]
```diff
@@ -53,25 +64,58 @@ const renderCreditCardInput = (formData, t) => (
...
-        placeholder={t('pages.payment_links.add_link.second_step.transfer_fees_label')}
-        options={buildInterestFeesOptions(formData.max_installments, t)}
-      />
-      {formData.free_installments > 0 && formData.credit_card && (
-        <InterestFees
-          name="interest_rate"
-          t={t}
-        />
+      {canChargeTransactionFee && (
+        <>
```
Comment (ThaisRobba) [inline packages/pilot/src/containers/PaymentLinks/List/PaymentLinkAdd/SecondStep/CreditCardInput.js:76]: Mudar aqui pra `<Fragment>` :)

File: packages/pilot/src/containers/PaymentLinks/List/PaymentLinkAdd/SecondStep/CreditCardInput.js [hunk 8]
```diff
@@ -53,25 +64,58 @@ const renderCreditCardInput = (formData, t) => (
...
+              placeholder={t('pages.payment_links.add_link.second_step.max_free_installments_label')}
+              options={buildInterestFeesOptions(12, t)}
+              value={formData.free_installments}
+            />
+            )
+          }
+        </>
+      )}
+      {!canChargeTransactionFee && (
+        <>
```
Comment (ThaisRobba) [inline packages/pilot/src/containers/PaymentLinks/List/PaymentLinkAdd/SecondStep/CreditCardInput.js:98]: El Fragmentito aqui tambien plisu

**Review by ThaisRobba — CHANGES_REQUESTED** [review]

File: packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js (hunk 6 above)
Comment (diegodsgarcia) [reply packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js:29]: É que nessa caso  eu teria que tratar também a visualização quando o freeInstallments === null (linha 19) e a visualização "default" (linha 77), que seria o detalhe de links sem o modelo de repasse de taxas. No final acho que ou cairia em um If ou em um Case mesmo.

File: packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js (hunk 5 above)
Comment (diegodsgarcia) [reply packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js:70]: Gostei da idéia do P, porém os spans vão ser necessários dentro dele para realizar as margins de espaçamento como o Vaguinho já tinha comentado sobre isso.
image

Comment (ThaisRobba) [reply packages/pilot/src/containers/PaymentLinks/Details/PaymentMethods/index.js]: Ahh sim. Não consigo pensar um jeito que fique mais limpo/claro e não acho que seria uma boa adicionar espaços nas strings, então acho show como está

**Comment by diegodsgarcia** [comment]
> @diegodsgarcia Dá uma olhadinha nos stories de `Containers|Payment Links/Details/PaymentMethods: WithInterestRate`
> e `Containers|Payment Links/Details/PaymentMethods: WithoutInterestRate` que ficou meio peculiar as mudanças 🤔

Tava realmente faltando o novo fluxo com Repasse. Coloquei lá agora 👍

File: packages/pilot/src/formatters/paymentLinkSpec.js [hunk 9]
```diff
@@ -23,6 +30,30 @@ const expiresIn = ({
...
+
+  return parseIntValue('max_installments', object)
+}
+
+const checkChargeTransactionAndSetInterestRate = (object) => {
+  if (object.chargeTransactionFee) {
+    return 0
+  }
+
+  return parseIntValue('interest_rate', object) || 0.01
```
Comment (vagnervst) [inline packages/pilot/src/formatters/paymentLinkSpec.js:54]: acho uma boa já retornarmos zero aqui, pois da forma que está, estamos setando pelo lojista uma taxa de juros sem avisa-lo, caso eu avance os steps sem preencher o interest_rate

```suggestion
  return parseIntValue('interest_rate', object) || 0
```

**Review by vagnervst — APPROVED** [review]
Só mais esse ponto e :shipit:

**Review by vagnervst — APPROVED** [review]

**Comment by ThaisRobba** [comment]
@diegodsgarcia No percy ainda parece meio estranho as telas com o repasse, é pepino no modo que está montado ou algo ficou peculiar mesmo? Tipo:
Captura de tela de 2020-09-21 11-22-19

**Comment by diegodsgarcia** [comment]
> @diegodsgarcia No percy ainda parece meio estranho as telas com o repasse, é pepino no modo que está montado ou algo ficou peculiar mesmo? Tipo:
> Captura de tela de 2020-09-21 11-22-19

Esse comportamento estranho no storybook é por que nós não utilizamos a lib do i18n. Nós só estamos fazendo um mock dela. Por isso deu esse comportamento esquisito :/ 

```js
const t = (sentence = '') => path(split('.', sentence), translations)
```

**Comment by marisamodolo** [comment]
Criando um link com 1x sem juros, o `Até` ficou com bold
image

**Review by marisamodolo — APPROVED** [review]
Tirando esse detalhe, tudo show :shipit:

**Comment by diegodsgarcia** [comment]
> Tirando esse detalhe, tudo show :shipit:

Esse detalhe acho que melhor infelizmente deixar passar, pois vai ter que adicionar mais uma condicional em todo o fluxo do plural que já está bem complicadinho xD

**Pull request merged**